codegen/smithy-aws-go-codegen/src/test/java/software/amazon/smithy/aws/go/codegen/customization/ApiGatewayExportsNullabilityExceptionIntegrationTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.loader.ModelAssembler;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.node.StringNode;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.DefaultTrait;
+
+public class ApiGatewayExportsNullabilityExceptionIntegrationTest {
+    private static final String PATH_PREFIX = "src/test/resources/software/amazon/smithy/aws/typescript/codegen/customization/apigw_exports";
+    private static final String EXPECTED_PATH_PREFIX = "src/main/resources/software/amazon/smithy/aws/go/codegen/customization/apigw_exports_exceptions";
+
+    @Test
+    public void test() throws IOException {
+        ApiGatewayExportsNullabilityExceptionIntegration integration = new ApiGatewayExportsNullabilityExceptionIntegration();
+        GoSettings settings = new GoSettings();
+        settings.setService(ShapeId.from("com.amazonaws.medialive#MediaLive"));
+        Model modelToPreprocess = new ModelAssembler()
+                .addImport(Path.of(PATH_PREFIX, "medialive.json").toAbsolutePath())
+                .putProperty(ModelAssembler.ALLOW_UNKNOWN_TRAITS, true)
+                .assemble()
+                .unwrap();
+        Model preprocessedModel = integration.preprocessModel(modelToPreprocess, settings);
+        assertEquals("",
```

Comment:
A string default isn't something that I'd expect to be impacted by this customization. I think we're looking for numbers and booleans. But maybe it's good to ensure unrelated shapes aren't impacted too, like string defaults.